src/app/pages/directoryservice/activedirectory/activedirectory.component.ts

```diff
@@ -454,9 +453,33 @@ export class ActiveDirectoryComponent {
     return this.ws.call('activedirectory.update', [body]);
   }
 
-  afterSubmit(value) {
+  
+  responseOnSubmit(value) {
+    this.entityEdit.formGroup.controls['kerberos_principal'].setValue(value.kerberos_principal);
+    this.entityEdit.formGroup.controls['kerberos_realm'].setValue(value['kerberos_realm']);
+
     if (value.enable) {
       this.adStatus = true;
     }
+
+    this.rest.get("directoryservice/kerberosrealm", {}).subscribe((res) => {
+      this.kerberos_realm = _.find(this.fieldConfig, {name : 'kerberos_realm'});
+      this.kerberos_realm.options.length = 0;
```

Comment:
I didn't know you could clear an array like this, but it looks like it should work lol